### PR TITLE
feat: Add immediate option to `ReplicationSource`

### PR DESCRIPTION
**Pull request by onedr0p**
**Describe what this PR does**

Adds immediate option to `ReplicationSource` - immediate can be used to stop replication when a new ReplicationSource is created. Defaults to "true".

**Is there anything that requires special attention?**

I am unable to run `make test` (Apple Silicon issue?) locally so looking for some feedback. This also could use some unit tests if required, however I am unable to write & test due to that issue until I find another way.

**Related issues:**
https://github.com/backube/volsync/issues/627

### CODE REVIEW

File: controllers/mover/restic/builder.go [hunk 1]
```diff
@@ -119,6 +119,8 @@ func (rb *Builder) FromSource(client client.Client, logger logr.Logger,
 	saHandler := utils.NewSAHandler(client, source, isSource, privileged,
 		source.Spec.Restic.MoverServiceAccount)
 
+	isImmediate := source.Spec.Immediate == nil
```
Comment (uhthomas) [inline controllers/mover/restic/builder.go:122]:
```suggestion
	isImmediate := source.Spec.Immediate != nil
```

?

Comment (uhthomas) [reply controllers/mover/restic/builder.go:122]: Isn't this variable a boolean, why is it being compared to nil?

Comment (onedr0p) [reply controllers/mover/restic/builder.go:122]: The instantiation of `Immediate` is a pointer, so here I am making the default `true`. I couldn't see another way to do this so I am all ears if there is a better way.

Comment (uhthomas) [reply controllers/mover/restic/builder.go:122]: I see, but the actual value is never consumed anywhere. It should be something like:

```suggestion
	isImmediate := true
	if source.Spec.Immediate != nil {

		isImmediate = * source.Spec.Immediate

	
}
```

**Comment by tesshuflower** [comment]
@onedr0p thanks for this PR, however I think if we do this there are a few updates we'd need to do:

- If we add immediate to replicationsource I'd prefer to also add to replicationdestination to be consistent
- Does it make sense to add `immediate` to spec.trigger somewhere? And potentially even call this something like `spec.trigger.scheduleImmediate`?  Since I think this will not apply (and should be ignored) by manual triggers.
- I think the actual implementation will need to be modified to set status/modify the scheduling algorithm to make sure we skip this initial sync if immediate is false.  The way it's done in the PR prevents the job from starting - which would be the same effect as creating a replicationsource with paused=true.  The problem with this is everything else will still happen immediately - i.e. snapshot/clone will be taken of the pvc immediately, and that pvc from snap/clone will be of the PVC at the time when the replicationsource is created.  I think to do this as you intend, we'd need to modify the scheduling to not sync at all at the start which is a little more involved as we don't want to risk breaking the scheduling.  

I'll discuss with the rest of the team to see if this does sound like a valid direction to go - I will try to take a look at the scheduling when I have a moment and see if it's something that could be done relatively easily.

**Comment by onedr0p** [comment]
Closing PR, will cycle back if needed after the volume populator is released.

**Pull request closed without merging**